app/javascript/src/lib/codeActions.js

```diff
@@ -0,0 +1,96 @@
+import { Decoration, ViewPlugin, ViewUpdate, WidgetType } from "@codemirror/view"
+
+/**
+ * @typedef CodeAction
+ * @property {number} position
+ * @property {string} label
+ * @property {Function} run
+ */
+
+/**
+ * @typedef {(update: ViewUpdate) => CodeAction[]} CodeActionProvider
+ */
+
+class CodeActionLightBulbWidget extends WidgetType {
+  constructor(actions) {
+    super()
+
+    this.actions = actions
+  }
+
+  toDOM() {
+    const container = document.createElement("span")
+    container.className = "button cm-codeActionsLightbulb"
+    container.textContent = "💡"
+
+    const dropdown = document.createElement("select")
```

Comment:
Why is this a select with options rather than a more regular div with buttons inside?

Reply:
Easier to implement 😛. Also better for mobile, I guess.

If it looks too ugly I guess I can try and change it.

Reply:
Fair enough!